lidar.py

```diff
@@ -0,0 +1,77 @@
+import uart
+import math
+
+DATA_HEADER_LOWER = 'aa'
+DATA_HEADER_UPPER = '5500'
+
+BAUD = 115200
+UART = uart.UART_0
+
+class Lidar():
+    def __init__(self, uart):
+        self.uart = uart
+
+    def readData(self):
+        self.uart.flush()
+        headerReceived = False
+        while(headerReceived is False):
+            while (self.uart.read(1).hex() != DATA_HEADER_LOWER):
+                pass
+            if (self.uart.read(2).hex() == DATA_HEADER_UPPER):
+                headerReceived = True
+
+        self.numVals = int(self.uart.read(1).hex(), 16)
+
+        fsaL = self.uart.read(1).hex()
+        fsaU = self.uart.read(1).hex()
+        self.fsa = int(fsaU + fsaL, 16)
+        
+        lsaL = self.uart.read(1).hex()
+        lsaU = self.uart.read(1).hex()
+        self.lsa = int(lsaU + lsaL, 16)
+
+        csL = self.uart.read(1).hex()
+        csU = self.uart.read(1).hex()
+        self.cs = int(csU + csL, 16)
+
+        self.rawData = self.uart.read(self.numVals * 2).hex()
+        self.dist = [0] * self.numVals
+        self.ang =  [0] * self.numVals
+
+    def dataCorrection(self):
+        checksum = 0x55AA
+        checksum ^= self.fsa
+
+        startAng = (self.fsa >> 1) / 64
+        endAng   = (self.lsa >> 1) / 64
+
+        diff = endAng - startAng
+        #if diff < 0:
+        #    diff += 360
```

Comment:
Is there a reason this is commented it out? Just checking

Reply:
I found a sample code that had the line, but the data sheet said nothing about it. I don't think it's needed but i left it there just in case